WorkspaceServer/Servers/Roslyn/CompilationUtility.cs

```diff
@@ -62,6 +68,31 @@ bool IsAvailable()
             }
         }
 
+        public static async Task DoWhenFileAvailable(
+            this FileInfo file, Action action)
+        {
+            if (file == null)
+            {
+                return;
+            }
+            const int waitAmount = 100;
+            var attemptCount = 1;
+            while (file.Exists && attemptCount <= 10)
```

Comment:
Can we do `await file.WaitForFileAvailable()` and then perform the action

Reply:
`await file.WaitForFileAvailable()` check for exclusive read access, this one attempts an action that may require different access.